### PR TITLE
[5.7] Add missing Arrayable typehints

### PR DESCRIPTION
**Pull request by bonzai**
This is continuation of #25669:

> It's possible to pass data to view using object that implements \Illuminate\Contracts\Support\Arrayable contract.

- https://github.com/laravel/framework/blob/5.7/src/Illuminate/View/Factory.php#L237

### CODE REVIEW

**Comment by TBlindaruk** [comment]
I`m not sure if it is really needed since it also can be `Jsonable` \ `JsonSerializable` |

Please see
Http/JsonResponse.php setData

**Comment by bonzai** [comment]
`shouldBeJson` is related to content, not a view data.

File: src/Illuminate/Support/Facades/Response.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 /**
  * @method static \Illuminate\Http\Response make(string $content = '', int $status = 200, array $headers = [])
- * @method static \Illuminate\Http\Response view(string $view, array $data = [], int $status = 200, array $headers = [])
+ * @method static \Illuminate\Http\Response view(string $view, \Illuminate\Contracts\Support\Arrayable | array $data = [], int $status = 200, array $headers = [])
```
Comment (lucasmichot) [inline src/Illuminate/Support/Facades/Response.php]: There should be no space around the `|`

File: src/Illuminate/Support/Facades/View.php [hunk 2]
```diff
@@ -4,8 +4,8 @@
 
 /**
  * @method static bool exists(string $view)
- * @method static \Illuminate\Contracts\View\View file(string $path, array $data = [], array $mergeData = [])
- * @method static \Illuminate\Contracts\View\View make(string $view, array $data = [], array $mergeData = [])
+ * @method static \Illuminate\Contracts\View\View file(string $path, \Illuminate\Contracts\Support\Arrayable | array $data = [], array $mergeData = [])
```
Comment (lucasmichot) [inline src/Illuminate/Support/Facades/View.php]: same here

File: src/Illuminate/Support/Facades/Response.php (hunk 1 above)
Comment (bonzai) [reply src/Illuminate/Support/Facades/Response.php]: It doesn't matter. Moreover, there are spaces around `|` in other facades too:
- https://github.com/laravel/framework/blob/5.7/src/Illuminate/Support/Facades/Event.php
- https://github.com/laravel/framework/blob/5.7/src/Illuminate/Support/Facades/Gate.php
- https://github.com/laravel/framework/blob/5.7/src/Illuminate/Support/Facades/Lang.php
- https://github.com/laravel/framework/blob/5.7/src/Illuminate/Support/Facades/Validator.php

I suggest you can create a new PR to remove them.

**Comment by sisve** [comment]
This is a breaking change. You have no idea if all the implementations of ResponseFactory in third-party packages also support Arrayable, or if they are built to only support array. Same thing applies to the Router contract.

**Comment by taylorotwell** [comment]
Needs better formatting.

File: src/Illuminate/Support/Facades/Response.php (hunk 1 above)
Comment (moderndevelopment) [reply src/Illuminate/Support/Facades/Response.php]: Just because there are mistakes in other parts of the framework doesn't mean that it doesn't matter.

**Pull request closed without merging**